agent/acl_test.go

```diff
@@ -163,6 +164,9 @@ func (a *TestACLAgent) Stats() map[string]map[string]string {
 func (a *TestACLAgent) ReloadConfig(_ consul.ReloadableConfig) error {
 	return fmt.Errorf("Unimplemented")
 }
+func (a *TestACLAgent) ResourceServiceClient() pbresource.ResourceServiceClient {
```

Comment:
Do we need this here? I didn't see this function is referenced anywhere.

Reply:
Yeah to resolve [type error](https://github.com/hashicorp/consul/actions/runs/5490422390/jobs/10005841647#step:5:232)